src/solve/normalization.md

```diff
@@ -3,32 +3,80 @@
 With the new solver we've made some fairly significant changes to normalization when compared
 to the existing implementation.
 
-We now differentiate between "shallow normalization" and "deep normalization".
-"Shallow normalization" normalizes a type until it is no-longer a potentially normalizeable alias;
-it does not recurse into the type. "deep normalization" replaces all normalizeable aliases in a
-type with its underlying type.
-
-The old trait solver currently always deeply normalizes via `Projection` obligations.
-This is the only way to normalize in the old solver. By replacing projections with a new
-inference variable and then emitting `Projection(<T as Trait>::Assoc, ?new_infer)` the old
-solver successfully deeply normalizes even in the case of ambiguity. This approach does not
-work for projections referencing bound variables.
-
-## Inside of the trait solver
-
-Normalization in the new solver exclusively happens via `Projection`[^0] goals.
-This only succeeds by first normalizing the alias by one level and then equating
-it with the expected type. This differs from [the behavior of projection clauses]
-which can also be proven by successfully equating the projection without normalizating.
-This means that `Projection`[^0] goals must only be used in places where we
-*have to normalize* to make progress. To normalize `<T as Trait>::Assoc`, we first create
-a fresh inference variable `?normalized` and then prove
-`Projection(<T as Trait>::Assoc, ?normalized)`[^0]. `?normalized` is then constrained to
-the underlying type. 
-
-Inside of the trait solver we never deeply normalize. we only apply shallow normalization
-in [`assemble_candidates_after_normalizing_self_ty`] and inside for [`AliasRelate`]
-goals for the [`normalizes-to` candidates].
+We now differentiate between "one-step normalization", "structural normalization" and
+"deep normalization".
+
+## One-step normalization
+
+One-step normalization is implemented via `NormalizesTo` goals. Unlike other goals
+in the trait solver, `NormalizesTo` always expects the term to be an unconstrained
+inference variable[^opaques]. Think of it as a function, taking an alias as input
+and returning its underlying value. If the alias is rigid, `NormalizesTo` fails and
+returns `NoSolution`.
```

Comment:
when does it happen that the alias is rigid?

Reply:
extended that section to explain taht